tests/apps/pubsub-bulk-subscriber/app.go

```diff
@@ -0,0 +1,577 @@
+			// Log only
+			log.Printf("(%s) Error extracing JSON from raw event in bulk entry %s is: %v", reqID, entry.EntryID, err)
+		} else {
+			log.Printf("(%s) Output of JSON from raw event in bulk entry %s is: %v", reqID, entry.EntryID, actualMsg)
+			entryDataStr = actualMsg
+		}
+
+		appMsg := AppBulkMessageEntry{
+			EntryID:  entry.EntryID,
+			EventStr: entryDataStr,
+		}
+		finalMsgs[i] = appMsg
+		log.Printf("(%s) output at index: %d, entry id:'%s' is: '%s':", reqID, i, entry.EntryID, entryData)
+	}
+	return finalMsgs, nil
+}
+
+// the test calls this to get the messages received
+func getReceivedMessages(w http.ResponseWriter, r *http.Request) {
+	reqID, ok := r.Context().Value("reqid").(string)
+	if reqID == "" || !ok {
+		reqID = "s-" + uuid.New().String()
+	}
+
+	response := receivedMessagesResponse{
+		ReceivedByTopicRawSub:     unique(receivedMessagesSubRaw.List()),
+		ReceivedByTopicCESub:      unique(receivedMessagesSubCE.List()),
+		ReceivedByTopicRawBulkSub: unique(receivedMessagesBulkRaw.List()),
+		ReceivedByTopicCEBulkSub:  unique(receivedMessagesBulkCE.List()),
+	}
+
+	log.Printf("getReceivedMessages called. reqID=%s response=%s", reqID, response)
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(response)
+}
+
+// setDesiredResponse returns an http.HandlerFunc that sets the desired response
+// to `resp` and logs `msg`.
+func setDesiredResponse(resp respondWith, msg string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		lock.Lock()
+		defer lock.Unlock()
+		log.Print(msg)
+		desiredResponse = resp
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+// handler called for empty-json case.
+func initializeHandler(w http.ResponseWriter, _ *http.Request) {
+	initializeSets()
+	w.WriteHeader(http.StatusOK)
+}
+
+// initialize all the sets for a clean test.
+func initializeSets() {
+	// initialize all the sets
+	receivedMessagesSubRaw = sets.NewString()
+	receivedMessagesSubCE = sets.NewString()
+	receivedMessagesBulkRaw = sets.NewString()
+	receivedMessagesBulkCE = sets.NewString()
+}
+
+// appRouter initializes restful api router
+func appRouter() *mux.Router {
+	log.Printf("Called appRouter()")
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Log requests and their processing time
+	router.Use(utils.LoggerMiddleware)
+
+	router.HandleFunc("/", indexHandler).Methods("GET")
+
+	router.HandleFunc("/getMessages", getReceivedMessages).Methods("POST")
+	router.HandleFunc("/set-respond-success",
+		setDesiredResponse(respondWithSuccess, "set respond with success")).Methods("POST")
+	router.HandleFunc("/set-respond-success-bulk",
+		setDesiredResponse(respondWithSuccessBulk, "set respond with success for bulk")).Methods("POST")
+	router.HandleFunc("/set-respond-error",
+		setDesiredResponse(respondWithError, "set respond with error")).Methods("POST")
+	router.HandleFunc("/set-respond-retry",
+		setDesiredResponse(respondWithRetry, "set respond with retry")).Methods("POST")
```

Comment:
unused

Reply:
Agreed, will do it separately under https://github.com/dapr/dapr/issues/5271. Added over there in the list. Resolving this comment here.